Refresh() and managing Read() and subscriber updates. README.md entry

spellgen opened the pull request:
Removed "very" before "thin wrapper"  :)

Added the Refresh() function

Managing the for-select loop a little: I don't reply to Read() or update subscriptions the update channel is empty.

Commit:
Refresh() and managing Read() and subscriber updates. README.md entry

spellgen commented:
Hm this passes consistently on my home box

dcarbone committed:
Adding a print to see what the travis env sees.

spellgen commented:
Yeah so that is a little weird - there are no updates on c while that print statement executes.
It would be nice to have a second set of eyes on the ConfigManager handler (for-select), doing some non-standard things there to prioritize updates over reads and also pausing the loop while queueing updates.

dcarbone commented:
My guess, because it's a pointer. By the time the require happens the value might have changed.

The actual "fix" is the sleep call. I'm not sure it should be a pointer.

Total guess on my part.

dcarbone committed:
Some updates:

- No longer returning a pointer to the subscription channels.
- Adding "Unsubscribe" test

dcarbone committed:
No longer closing subscription channels in configurator test.  Workflow
warrants some discussion...

dcarbone committed:
Change TestUnsubscribe "ok" state a bit...

Commit:
added stop for prefix watcher
